### DISCUSSION TITLE
Write a workbook into a Buffer (example request)

### DISCUSSION THREAD

**Discussion by chunkjs** [exceljs/exceljs#2495] [Q&A]
Hello, 

There's an old issue requesting an example on how to get a workbook into a Buffer #20 , I tried using the StreamBuf but I'm currently getting an error stating that "StreamBuf is not a constructor".  Could you provide a recent example for achieving this?  Thanks in advance.

**Comment by Siemienik** [comment]
@chunkjs please share code with us

**Comment by chunkjs** [comment]
Thanks @Siemienik.  This is what I have:

```
import * as excel from 'exceljs';
import { StreamBuf} from 'exceljs/dist/es5/utils/stream-buf';
    let workBook = new excel.Workbook();
    workBook.addWorksheet("New Sheet");
    let workSheet = workBook.getWorksheet("New Sheet");
    workSheet.addRows([1,2,3,4]);

            var buf = new StreamBuf();//get an error here "StreamBuf is not a constructor"
                workBook.xlsx.write(buf)
                .then(function() {
                    var data = buf.toBuffer();
                    //do something with my buffer

                });
```

**Comment by chunkjs** [comment]
@Siemienik , in the end I want to convert the buffer to a base64 encoded string and send that as a response to my client.  The reason for this is that will send it back to my client through an AWS lambda

**Comment by purevdalai** [comment]
the same problem is here. Did you solve it?

**Comment by Siemienik** [comment]
did you try this way?
```js
const buffer = await workbook.xlsx.writeBuffer();
```

https://github.com/exceljs/exceljs#writing-xlsx

**Comment by oceanofmaya** [comment]
I needed to use this in a typescript project wherein I had to return the buffer as a base64 encoded string. The following worked for me. Note: Without the casting, my IDE errored out saying `Expected 0 arguments but got 1`

```
const fileBuffer = await workbook.xlsx.writeBuffer();
const base64String = (fileBuffer as Buffer).toString("base64");
```
